[1m============================= test session starts ==============================[0m
platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[1mcollecting ... [0m[1m
collected 1 item                                                               [0m

test/test_integration.py [INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-05-22-54-43-382334-PHR-57091
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [number_publisher-1]: process started with pid [57105]
[INFO] [number_processor-2]: process started with pid [57107]
test_communication_and_logic (test_integration.TestIntegration) ... [0m[INFO] [1754405683.426759387] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[number_processor-2] [INFO] [1754405683.500070244] [number_processor]: 数字处理节点已启动，等待接收数字...
[number_publisher-1] [INFO] [1754405683.500070244] [number_publisher]: 数字发布节点已启动，将从1开始发布...
[number_publisher-1] [INFO] [1754405684.495113232] [number_publisher]: 发布: 1
[number_processor-2] [INFO] [1754405684.495572716] [number_processor]: 收到: 1
[number_processor-2] [INFO] [1754405684.496163990] [number_processor]: 发布处理后结果: 2
[number_publisher-1] [INFO] [1754405685.495349494] [number_publisher]: 发布: 2
[number_processor-2] [INFO] [1754405685.495556084] [number_processor]: 收到: 2
[number_processor-2] [INFO] [1754405685.496121237] [number_processor]: 发布处理后结果: 4
[number_publisher-1] [INFO] [1754405686.495129870] [number_publisher]: 发布: 3
[number_processor-2] [INFO] [1754405686.495395825] [number_processor]: 收到: 3
[number_processor-2] [INFO] [1754405686.496093956] [number_processor]: 发布处理后结果: 6
[0m[INFO] [1754405686.497098409] [test_subscriber_node]: 检查接收到的消息。 Got: [2, 4, 6][0m
[0m[INFO] [1754405686.501478350] [test_subscriber_node]: 测试结束，销毁测试节点[0m
ok
test_processor_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754405686.510479251] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[0m[INFO] [1754405686.511025408] [test_subscriber_node]: 正在检查 processor 节点的启动日志...[0m
[0m[INFO] [1754405686.511729475] [test_subscriber_node]: Processor 节点启动正常。[0m
[0m[INFO] [1754405686.516117726] [test_subscriber_node]: 测试结束，销毁测试节点[0m
ok
test_publisher_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754405686.523816163] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[0m[INFO] [1754405686.524297018] [test_subscriber_node]: 正在检查 publisher 节点的启动日志...[0m
[0m[INFO] [1754405686.524783810] [test_subscriber_node]: Publisher 节点启动正常。[0m
[0m[INFO] [1754405686.528127240] [test_subscriber_node]: 测试结束，销毁测试节点[0m
ok

----------------------------------------------------------------------
Ran 3 tests in 3.138s

OK
[INFO] [number_processor-2]: sending signal 'SIGINT' to process[number_processor-2]
[INFO] [number_publisher-1]: sending signal 'SIGINT' to process[number_publisher-1]
[number_processor-2] Traceback (most recent call last):
[number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[number_processor-2]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_processor_node.py", line 32, in main
[number_processor-2]     rclpy.spin(node)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[number_processor-2]     executor.spin_once()
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[number_processor-2]     self._spin_once_impl(timeout_sec)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[number_processor-2]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[number_publisher-1] Traceback (most recent call last):
[number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher", line 33, in <module>
[number_publisher-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_publisher')())
[number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_publisher_node.py", line 25, in main
[number_publisher-1]     rclpy.spin(node)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[number_publisher-1]     executor.spin_once()
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[number_processor-2]     return next(self._cb_iter)
[number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[number_processor-2]     wait_set.wait(timeout_nsec)
[number_processor-2] KeyboardInterrupt
[number_publisher-1]     self._spin_once_impl(timeout_sec)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[number_publisher-1]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[number_publisher-1]     return next(self._cb_iter)
[number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[number_publisher-1]     wait_set.wait(timeout_nsec)
[number_publisher-1] KeyboardInterrupt
[ERROR] [number_processor-2]: process has died [pid 57107, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor'].
[ERROR] [number_publisher-1]: process has died [pid 57105, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher --ros-args -r __node:=number_publisher'].

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
[32m.[0m

- generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
[32m============================== [32m[1m1 passed[0m[32m in 3.26s[0m[32m ===============================[0m
